config/config.py

```diff
@@ -1,7 +1,13 @@
 from pathlib import Path
 
 import yaml
-from pydantic import BaseModel, FilePath
+from pydantic import BaseModel, DirectoryPath, FilePath
+
+
+class Config(BaseModel):
+    sql_script_path: FilePath
+    csv_folder: DirectoryPath
+    db_path: Path
```

Comment:
Good use of Pydantic to specify/validate paths (e.g. the the `csv_folder`). However, I would put these on a class where these attributes are used (e.g. `ReportGenerator`) instead of just creating a config object that is just a bag of values.